pixy/httpapi.go

```diff
@@ -131,6 +133,18 @@ func (as *HTTPAPIServer) handleProduce(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if isSync {
+		switch err := as.kafkaClient.Produce(
+			topic, toEncoderPreservingNil(key), sarama.StringEncoder(message)); err {
+		case nil:
+			w.WriteHeader(http.StatusOK)
+		case sarama.ErrUnknownTopicOrPartition:
+			http.Error(w, err.Error(), http.StatusNotFound)
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
This can be propagated to the end user. In most cases that would mean a Kafka catastrophic failure such that even after 3 retries a leader for a particular partition was not found. At this stage of development I would prefer seeing errors right up front in HTTP responses to make them more visible. When we approach the stage of public beta, then such internal detail can be hidden.